week7/src/FolderPage.js

```diff
@@ -40,44 +53,162 @@ function FolderPage() {
   const handlePrintFolderName = (name) => {
     setFolderName(name);
   };
+  const isShowModal = (linkAddModal) => {
+    setAddModal(linkAddModal);
+  };
 
+  const isShareModal = (e) => {
+    e.preventDefault();
+    isShowModal((prev) => ({
+      linkModal: false,
+      folderAddModal: false,
+      shareAddModal: true,
+    }));
```

Comment:
3가지 모달상태만 객체에 선언해놓으면 나머지 모달 상태를 잃어버리게 되요
`spread 연산자와`,`prev` 상태를 이용해서 앞선 상태를 먼저 할당해주고, 내가 바꾸고 싶은 상태를 주면 됩니다.

```suggestion
    isShowModal((prev) => ({
     ...prev,
     shareAddModal: true,
    }));
```